doc/designs/evp_skey.md

```diff
@@ -0,0 +1,182 @@
+EVP_SKEY Design
+===============
+
+We currently presume that symmetric keys are represnted as octet buffer. It may
+be not relevant for the situation of hardware-backed providers, such as
+PKCS#11 or TPM.
+
+This is a design of using opaque usage in symmetric encryption.
+
+It's quite obvious that EVP_SKEY will have to be modeled after EVP_PKEY insofar
+that it will have to have a cache of provider-specific instances of a key
+reference, for any provider that supports this key form.  It's also quite
+obvious that if the "origin" key is a reference to a provider-backed symmetric
+key that support exporting "a bunch of bytes", that can be cached and made
+unattached to any provider, and become useful to any current cipher.
```

Comment:
Considering the following discussion thread, this may need to be reformulated.

https://github.com/openssl/openssl/discussions/21355#discussioncomment-6755079

Reply:
I'm not sure I understand what you mean. There may be providers that don't support exporting the key as bunch of bytes (and, so, reuse for a different cipher) but I think we have enough ifs in this paragraph to cover this case...

Reply:
This part suggests that we would do the same export/import dance as with EVP_PKEY:

> that it will have to have a cache of provider-specific instances of a key
reference, for any provider that supports this key form.

However, that is not true as per the discussion thread.  That thread suggests that *if the provider allows exporting at all*, it would be as "a bunch of bytes", and we would cache that, unattached from any provider, rather than multiple provider references.

Reply:
What about this wording?
```
It's quite obvious that EVP_SKEY will have to be in general modeled after EVP_PKEY insofar.
If the provider allows key exporting, the internal representation is "a bunch of bytes", that can be cached and made unattached to any provider, and become useful to any current cipher supporting the symmetric key in byte form. 
In case when the export is impossible it will have to have a cache of provider-specific instances of a key
reference, for any provider that supports this key form. 
```

Reply:
Remove this:

> In case when the export is impossible it will have to have a cache of provider-specific instances of a key reference, for any provider that supports this key form.

If an export is impossible, *it's also impossible to cache it for other providers*

Reply:
Correct. So just
```
It's quite obvious that EVP_SKEY will have to be in general modeled after EVP_PKEY insofar.
If the provider allows key exporting, the internal representation is "a bunch of bytes", that 
can be cached and made unattached to any provider, and become useful to any current cipher 
supporting the symmetric key in byte form. 
```

Reply:
... yes, and with that, it's not so much modeled after EVP_PKEY any more.  That removes the first sentence, and requires that EVP_SKEY gets described as its own thing, without referencing EVP_PKEY.

Reply:
Update pushed